src/gcp/firestore.ts

```diff
@@ -149,7 +154,7 @@ export async function getDatabase(
   database: string,
   allowEmulator: boolean = false,
 ): Promise<Database> {
-  const apiClient = allowEmulator ? emuOrProdClient : prodOnlyClient;
+  const apiClient = getClient(allowEmulator);
```

Comment:
（offline discussion)

Not a big fan of the `allowEmulator` boolean + the env override. Can do a simple refactor to change them to take a `apiClient` override option.
If the client wants to use emulator, it can override.